acr_controller/service/acr_service.go

```diff
@@ -173,7 +173,10 @@ func (c *acrService) patchOperationSyncResultWithChangeRevision(ctx context.Cont
 
 func (c *acrService) getRevisions(ctx context.Context, a *application.Application) (string, string) {
 	if a.Status.History == nil || len(a.Status.History) == 0 {
-		// it is first sync operation, and we dont need detect change revision in such case
+		// it is first sync operation, and we have only current revision
+		if a.Operation != nil && a.Operation.Sync != nil {
```

Comment:
I guess you still don't cover case when history len = 1?

Reply:
Fixed